### ISSUE TITLE
Proper transaction result type

### ISSUE THREAD

**Issue by akorchyn** [near/near-api-rs#5]
Currently, developing an app requires `near-primitives` to interact with the result types of transactions as `near-api` propagates them.

As for me, it's pretty inconvenient, and would be better if we created some own wrapper type as `near-workspaces` [does](https://github.com/near/near-workspaces-rs/blob/05e3909e7839bd330fba714c724d83b560c91c95/workspaces/src/result.rs#L69)

**Comment by akorchyn** [comment]
I updated the issue as we have a lot of types coming from near-primitives when querying the chain.
It's mandatory to replace them with our types and not export them to have a more robust API overall.

The motivation is simple. We don't want to have a breaking release on each near-primitives release as it's not that convenient to use. To have that, `near-api` should have all the parts wrapped :)

**Comment by akorchyn** [comment]
The work on it will be done in `own-types` branch

**Comment by r-near** [comment]
To that end, it might also make sense to remove the dependency on `near-jsonrpc-client`. IMO it influences too much downstream. [near-fetch](https://github.com/ChaoticTempest/near-fetch) is similarly at the behest of this package, too.

**Comment by akorchyn** [comment]
Absolutely. We will need to cut all near core crates in favor new auto generated
